search.html

```diff
@@ -223,49 +231,163 @@
+        });
+
+        // collect tag counts
+        results.map(function(result) {
+          if (store[result.ref].tags !== undefined && store[result.ref].tags.length != 0) {
+            for (var i = 0; i < store[result.ref].tags.length; i++) {
+              if (tag_counts[store[result.ref].tags[i]] === undefined) {
+                tag_counts[store[result.ref].tags[i]] = 1;
+              } else {
+                tag_counts[store[result.ref].tags[i]] = tag_counts[store[result.ref].tags[i]] + 1;
+              }
+            }
+          }
+        });
+
+        // apply search filters and display in the UI
+        // search_type
+        if (search_type === undefined) {
+          $("#facet-type").empty();
+          for (var i = 0; i < types.length; i++) {
+            var type = types[i];
+            if (type_counts[type] !== undefined) {
+              var readable_type = type.split('_').join(' ');
+              var param_string = "?query=" + query + "&type=" + type;
+              var text_string = readable_type + " (" + type_counts[type] + ")"
+              $("#facet-type").append(
+                $('<a class="tag-btn">')
+                  .attr("href", param_string)
+                  .text(text_string)
+              );
+            }
+          }
+        } else { // type exists in the url; append removal btn
+          var url = window.location.href;
+          var altered_url = remove_url_param("type", url);
+          // var altered_url = remove_query_strings();
+          var readable_type = search_type.split('_').join(' ');
+          var text_string = readable_type + " ";
+          $(".applied-filters").empty().append(
+            $('<a class="tag-btn">')
+              .attr('href', altered_url)
+              .text(text_string).append(
+                $('<i class="fa fa-times" aria-hidden="true"></i>')
+              )
+          );
+        }
+
+        // search_tag
+        $("#facet-tag").empty();
+        if (tag_counts !== undefined
+            && !(typeof search_tag === 'string' || search_tag instanceof String)) {
+          for (var tag in tag_counts) {
+            var param_string = "?query=" + query + "&tag=" + tag;
+            var text_string = tag + " (" + tag_counts[tag] + ")"
+            $("#facet-tag").append(
+              $('<a class="tag-btn">')
+                .attr("href", param_string)
+                .text(text_string)
+            );
+          }
+        } else { // tag exists in the url; append removal btn
+          var url = window.location.href;
+          var altered_url = remove_url_param("tag", url);
+          var text_string = search_tag + " ";
+          $(".applied-filters").empty().append(
+            $('<a class="tag-btn">')
+              .attr('href', altered_url)
+              .text(text_string).append(
+                $('<i class="fa fa-times" aria-hidden="true"></i>')
+              )
+          );
+        }
+
+        // toggle visibility of search filters
```

Comment:
I read a little online and it doesn't seem that CSS could support conditional CSS based on child elements although it would otherwise be great for this use case. People suggested jQuery so I went with this slightly clunky way. http://stackoverflow.com/questions/2326499/apply-css-styles-to-an-element-depending-on-its-child-elements